torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2691,16 +2688,18 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collective(
   work->outputs_ =
       std::make_shared<std::vector<at::Tensor>>(std::move(outputs));
 
-  if (avoidRecordStreams) {
+  if (avoidRecordStreams && inputs.size() == 0) {
```

Comment:
Hi, I think here is check `inputs.size() > 0`, since I saw get `inputs[0]` in below, is that right?